### ISSUE TITLE
Inconsistent: Tensor repeat method is 1-indexed, not 0-indexed

### ISSUE THREAD

**Issue by atiyo** [pytorch/pytorch#6155]
On 0.3.1b0+2b47480 on macOS, built from source.

E.g.
`a = torch.Tensor([[5,2]])`
`a.repeat(2,0)`

errors out, but

`a.repeat(2,1)`
returns the 2x2 tensor that I would anticipate from the first attempt.

It appears that the repeat method is indexing dimensions from 1, not 0.

`torch.cat` indexes dimensions from 0. I thought the repeat method should be consistent.

**Comment by zou3519** [comment]
Could you please provide more detail? What do you mean by 1-indexed and 0-indexed? http://pytorch.org/docs/master/tensors.html?highlight=repeat#torch.Tensor.repeat

**Comment by atiyo** [comment]
Apologies. Accidentally posted the issue before typing anything! Added detail now.

**Comment by zou3519** [comment]
The arguments to repeat aren't indices or dimensions. They're the number of times you want to repeat a specific dimension.
```
a = torch.Tensor([[5,2]])
a.repeat(2,1)
```
Means you're repeating dimension 0 twice and dimension 1 once (so not at all).

**Comment by SsnL** [comment]
@atiyo The [doc](http://pytorch.org/docs/master/tensors.html#torch.Tensor.repeat) clearly says that `.repeat` takes in sizes. There is no dim argument. This is the expected behavior.

**Comment by atiyo** [comment]
Apologies. Closing this.
